src/sklearn_evaluation/models.py

```diff
@@ -0,0 +1,430 @@
+from sklearn.metrics import accuracy_score
+from sklearn.metrics import confusion_matrix, auc
+import numpy as np
+import re
+from sklearn_evaluation.evaluator import _gen_ax
+from sklearn_evaluation import plot
+from sklearn_evaluation.model_heuristics.utils import (
+    check_array_balance,
+    get_model_prediction_time,
+    get_roc_auc,
+    Range,
+)
+from sklearn_evaluation.model_heuristics.model_heuristics import ModelHeuristics
+
+
+class ReportSection:
+    """
+    Section to include in report
+    """
+
+    def __init__(self, key, include_in_report=True):
+        self.report_section = dict(
+            {
+                "guidelines": [],
+                "title": key.replace("_", " "),
+                "include_in_report": include_in_report,
+                "is_ok": False,
+            }
+        )
+        self.key = key
+
+    def append_guideline(self, guideline):
+        """
+        Add guideline to section
+
+        Parameters
+        ----------
+        guideline : str
+            The guideline to add
+        """
+        self.report_section["guidelines"].append(guideline)
+
+    def get_dict(self) -> dict:
+        """
+        Return dict of the section
+        """
+        return self.report_section
+
+    def set_is_ok(self, is_ok):
+        """
+        Set if the reported test is valid
+        """
+        self.report_section["is_ok"] = is_ok
+
+    def set_include_in_report(self, include):
+        """
+        Set if should include this section in the report
+        """
+        self.report_section["include_in_report"] = include
+
+
+class ModelEvaluator(ModelHeuristics):
+    """
+    Generates model evaluation report
```

Comment:
let's briefly document what are the rules implemented so the users know what to expect. e.g.:

- If minority class is < 5%, then a class imbalance warning is displayed
- if ... then ...